features/bsm/rest/api/src/main/java/org/opennms/web/rest/v2/bsm/model/edge/ApplicationEdgeRequestDTO.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.rest.v2.bsm.model.edge;
+
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+@XmlRootElement(name="application-edge")
+public class ApplicationEdgeRequestDTO extends AbstractEdgeRequestDTO {
+
+    private Integer applicationId;
+
+    @XmlElement(name="application-id",required = true)
+    public Integer getApplicationId() {
+        return applicationId;
+    }
+
+    public void setApplicationId(Integer applicationId) {
+        this.applicationId = applicationId;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) { return false; }
+        if (!(obj instanceof IpServiceEdgeRequestDTO)) { return false; }
```

Comment:
Looks like the class is wrong in this instanceof comparison. Here is the IntelliJ generated equals if you want to just use that
```
    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        if (!super.equals(o)) return false;
        ApplicationEdgeRequestDTO that = (ApplicationEdgeRequestDTO) o;
        return Objects.equals(applicationId, that.applicationId);
    }
```
```suggestion
        if (!(obj instanceof ApplicationEdgeRequestDTO)) { return false; }
```